solr/core/src/java/org/apache/solr/cloud/api/collections/CreateAliasCmd.java

```diff
@@ -136,6 +139,16 @@ private Instant parseStart(String str, TimeZone zone) {
     return start;
   }
 
+  private void checkPreemptiveCreateWindow(String str) {
+    if (StringUtils.isNotBlank(str)) {
+      try {
+        new DateMathParser().parseMath( "-" + str);
```

Comment:
I thought about allowing the user to specify the '-' but it just feels weird to think about. You wind up reading it like "Preemptively create a collection at -3 hours in the future relative to the last collection" or "Preemptively create a collection when it's needed -3 hours" rather than the much more natural thought/sentence "Preemptively create a collection 3 hours before it's needed." So I felt the API "read" better this way.

Reply:
I do like that we don't make the user specify it -- that's nice.  I agree it'd be confusing to know when to use +/- if we made users specify it.  It'd be nice if other parts of Solr had this convenience, like range facet gap.  I'm just saying we should internally convert to full DateMath eagerly; otherwise it's this partial datemath type of string and thus it's inaccurate to even say, via the variable name, that it's date math.  Or... maybe we allow DateMath to have the leading operator be optional defaulting to '+', and thus it's accurate to say that "1SEC" is date math?